lk2nd/device/keys.c

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright (c) 2022 Nikita Travkin <nikita@trvn.ru> */
+
+#include <debug.h>
+#include <libfdt.h>
+#include <stdlib.h>
+#include <string.h>
+#include <target.h>
+
+#include <lk2nd/gpio.h>
+#include <lk2nd/keys.h>
+
+#include "device.h"
+
+#define LK2ND_MAX_KEYS 4
+
+struct lk2nd_keymap {
+	uint32_t keycode;
+	struct gpio_desc gpio;
+};
+
+static struct lk2nd_keymap keymap[LK2ND_MAX_KEYS] = {0};
+
+/**
+ * lk2nd_keys_pressed() - Check if the key is pressed.
+ * @keycode: Keycode to check.
+ *
+ * The method will first check the override keymap
+ * and try to find a gpio attached to the keycode.
+ * If there is no override key for the code, the
+ * default target-specific key checks will be used
+ * for certain keys.
+ *
+ * Returns: 1 if the key is pressed, 0 if not.
+ *
+ * If the key is absent, 0 will be returned.
+ */
+int lk2nd_keys_pressed(uint32_t keycode)
+{
+	int i = 0, st;
+	bool found = false;
+
+	for (i = 0; i < LK2ND_MAX_KEYS; ++i) {
+		if (!keymap[i].keycode)
+			break;
+
+		if (keymap[i].keycode == keycode) {
+			found = true;
```

Comment:
This seems to handle the case that you might define multiple GPIOs with the same keycode. Is this intended? Might be worth a comment somewhere (possibly just in the kerneldoc above)